lib/scholar/spatial/distances.ex

```diff
@@ -0,0 +1,271 @@
+      iex> Scholar.Spatial.Distances.euclidean(x, y)
+      ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+  """
+  @spec euclidean(Nx.t(), Nx.t()) :: Nx.t()
+  defn euclidean(x, y) do
+    assert_shape!(x, y)
+    diff = x - y
+
+    cond do
+      Nx.all(diff == 0) ->
+        Nx.tensor(0.0)
+
+      true ->
+        diff
+        |> Nx.LinAlg.norm()
+        |> Nx.as_type({:f, 32})
+    end
+  end
+
+  @doc """
+  Squared euclidean distance.
+
+  $$
+  D(x, y) = \\sum_i (x_i - y_i)^2
+  $$
+
+  ## Examples
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([3, 2])
+      iex> Scholar.Spatial.Distances.squared_euclidean(x, y)
+      #Nx.Tensor<
+        f32
+        4.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2])
+      iex> Scholar.Spatial.Distances.squared_euclidean(x, y)
+      #Nx.Tensor<
+        f32
+        0.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2, 3])
+      iex> Scholar.Spatial.Distances.squared_euclidean(x, y)
+      ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+  """
+  @spec squared_euclidean(Nx.t(), Nx.t()) :: Nx.t()
+  defn squared_euclidean(x, y) do
+    assert_shape!(x, y)
+
+    x
+    |> Nx.subtract(y)
+    |> Nx.power(2)
+    |> Nx.sum()
+    |> Nx.as_type({:f, 32})
+  end
+
+  @doc """
+  Manhattan, taxicab, or l1 distance.
+
+  $$
+  D(x, y) = \\sum_i |x_i - y_i|
+  $$
+
+  ## Examples
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([3, 2])
+      iex> Scholar.Spatial.Distances.manhattan(x, y)
+      #Nx.Tensor<
+        f32
+        2.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2])
+      iex> Scholar.Spatial.Distances.manhattan(x, y)
+      #Nx.Tensor<
+        f32
+        0.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2, 3])
+      iex> Scholar.Spatial.Distances.manhattan(x, y)
+      ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+  """
+  @spec manhattan(Nx.t(), Nx.t()) :: Nx.t()
+  defn manhattan(x, y) do
+    assert_shape!(x, y)
+
+    x
+    |> Nx.subtract(y)
+    |> Nx.abs()
+    |> Nx.sum()
+    |> Nx.as_type({:f, 32})
+  end
+
+  @doc """
+  Chebyshev or l-infinity distance.
+
+  $$
+  D(x, y) = \\max_i |x_i - y_i|
+  $$
+
+  ## Examples
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([3, 2])
+      iex> Scholar.Spatial.Distances.chebyshev(x, y)
+      #Nx.Tensor<
+        f32
+        2.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2])
+      iex> Scholar.Spatial.Distances.chebyshev(x, y)
+      #Nx.Tensor<
+        f32
+        0.0
+      >
+
+      iex> x = Nx.tensor([1, 2])
+      iex> y = Nx.tensor([1, 2, 3])
+      iex> Scholar.Spatial.Distances.chebyshev(x, y)
+      ** (ArgumentError) expected input shapes to be equal, got {2} != {3}
+  """
+  @spec chebyshev(Nx.t(), Nx.t()) :: Nx.t()
+  defn chebyshev(x, y) do
```

Comment:
There's also a variant of Nx.LinAlg.norm for this and the one below IIRC

Reply:
Yep, `Nx.LinAlg.norm(ord: :inf)` works here.